src/features/courses/course-controller.ts

```diff
@@ -3027,15 +3027,21 @@ class CourseController {
         let message = '';
         if (user.roleId === Role.PROFESSOR || user.roleId === Role.ADMIN) return {userCanViewQuestion: true, message};
         const question = await courseRepository.getQuestion({ id: questionId });
-        const topic = await question.getTopic();
-        if (topic.topicTypeId === 1) {
-            if (topic.startDate.toMoment().isAfter(moment())) {
-                message = `${topic.name} hasn't started yet.`;
-                return { userCanViewQuestion: false, message };
-            } else {
-                return { userCanViewQuestion: true, message };
-            }
-        } else if (topic.topicTypeId === 2) {
+        let topic = await question.getTopic();
+        const topicOverride = await courseRepository.getStudentTopicOverride({userId: user.id, topicId: topic.id});
+        if (!_.isNil(topicOverride)) {
+            topic = topic.getWithOverrides(topicOverride) as CourseTopicContent;
```

Comment:
This is casting to an invalid type, the typing will say you have functions and attributes that will fail

i.e. if you try to do topic.getQuestions typescript will say ok and react will crash

Reply:
We should make the type of topic be CourseTopicContentInterface

Reply:
const dbTopic = await question.getTopic();
let topic: CourseTopicContentInterface = dbTopic;
topic = dbTopic.getWithOverrides(topicOverride);